### PR TITLE
Dynamically choose python binary

### PR DESCRIPTION
**Pull request by bmw**
Different approach to solving the problem discussed in #681.

Thoughts @jmhodges, @kuba?

### CODE REVIEW

**Comment by bmw** [comment]
Travis glitch caused first failure. Possibly caused by race condition with `time.sleep(4)` before verifying setup (note this has `XXX` comment and is only used in test code).

File: letsencrypt/plugins/manual.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ class ManualAuthenticator(common.Plugin):
 cd {root}/public_html
 echo -n {validation} > {response.URI_ROOT_PATH}/{encoded_token}
 # run only once per server:
-python -c "import BaseHTTPServer, SimpleHTTPServer; \\
+$(command -v python2 || command -v python2.7 || command -v python2.6) -c \\
```
Comment (kuba) [inline letsencrypt/plugins/manual.py]: Have you considered using `type -P` instead (looks also for keywords, aliases, functions)?

**Comment by kuba** [comment]
LGTM modulo comment above

File: letsencrypt/plugins/manual.py (hunk 1 above)
Comment (bmw) [inline letsencrypt/plugins/manual.py]: I looked at `which`, `hash`, `type`, and `command` but `command` seemed to work best. According to [this](http://pubs.opengroup.org/onlinepubs/009696899/utilities/command.html), `command -v` also looks at aliases and functions (but not keywords). `command -v` also seemed the most portable as `type -P` doesn't work in shells such as zsh for instance.

**Comment by jdkasten** [comment]
Thanks!

**Pull request merged**